### PR TITLE
Log events fixes

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

File: vmhost/contexts/output.go [hunk 1]
```diff
@@ -2,8 +2,8 @@ package contexts
 
 import (
 	"bytes"
-	"encoding/hex"
 	"errors"
+	"github.com/multiversx/mx-chain-vm-common-go/parsers"
```
Comment (iulianpascalau) [inline vmhost/contexts/output.go:6]: go imports

File: vmhost/contexts/asyncLocal.go [hunk 2]
```diff
@@ -195,33 +195,9 @@ func (context *asyncContext) executeSyncHalfOfBuiltinFunction(asyncCall *vmhost.
...
-	err error,
-	destinationReturnCode vmcommon.ReturnCode,
-) {
-
+func (context *asyncContext) finishAsyncLocalCallbackExecution() {
 	runtime := context.host.Runtime()
-
 	runtime.GetVMInput().GasProvided = 0
-
-	// if vmOutput == nil {
```
Comment (iulianpascalau) [inline vmhost/contexts/asyncLocal.go:210]: so I guess we no longer need the commented code :D

File: vmhost/contexts/output.go [hunk 3]
```diff
@@ -387,9 +393,44 @@ func (context *outputContext) Transfer(destination []byte, sender []byte, gasLim
 
 	logOutput.Trace("transfer value added")
 
+	function, args, errNotCritical := context.callArgsParser.ParseData(string(input))
+	if !isBackTransfer && (errNotCritical != nil || !core.IsSmartContractAddress(destination) || gasLimit == 0) {
```
Comment (iulianpascalau) [inline vmhost/contexts/output.go:397]: could have been extracted in a bool var

File: vmhost/hostCore/execution.go [hunk 4]
```diff
@@ -957,6 +971,13 @@ func (host *vmHost) ExecuteESDTTransfer(transfersArgs *vmhost.ESDTTransfersArgs,
 		}
 	}
 
+	if len(transfersArgs.Function) > 0 {
+		esdtTransferInput.Arguments = append(esdtTransferInput.Arguments, []byte(transfersArgs.Function))
+	}
+	if len(transfersArgs.Arguments) > 0 {
+		esdtTransferInput.Arguments = append(esdtTransferInput.Arguments, transfersArgs.Arguments...)
```
Comment (iulianpascalau) [inline vmhost/hostCore/execution.go:978]: Is this ok? What happens if the esdtTransferInput struct contains the Function field set to empty and it contains a list of arguments? Should we prevent that situation? If yes, we need to move L977-L979 after L975

Comment (sasurobert) [reply vmhost/hostCore/execution.go:978]: that is not possible.

**Review by CostinCarabas — DISMISSED** [review]

**Review by iulianpascalau — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by CostinCarabas — APPROVED** [review]

File: vmhost/common.go [hunk 5]
```diff
@@ -225,7 +225,8 @@ type AsyncGeneratedCall struct {
 }
 
 // OldAsyncContext is a structure containing a group of async calls and a callback
-//  that should be called when all these async calls are resolved
+//
+//	that should be called when all these async calls are resolved
```
Comment (camilbancioiu) [inline vmhost/common.go:229]: Indentation changed?

**Pull request merged**